flytekit/core/type_engine.py

```diff
@@ -488,9 +509,83 @@ def get_literal_type(self, t: Type[T]) -> LiteralType:
 
         ts = TypeStructure(tag="", dataclass_type=literal_type)
 
-        return _type_models.LiteralType(simple=_type_models.SimpleType.STRUCT, metadata=schema, structure=ts)
+        return _type_models.LiteralType(simple=_type_models.SimpleType.JSON, metadata=schema, structure=ts)
+
+    # We use UTF-8 as the default serialization format for JSON
+    def to_json(self, ctx: FlyteContext, python_val: T, python_type: Type[T], expected: LiteralType) -> Literal:
+        if isinstance(python_val, dict):
+            json_str = json.dumps(python_val)
+            json_bytes = json_str.encode("UTF-8")
+            return Literal(scalar=Scalar(json=Json(value=json_bytes, serialization_format="UTF-8")))
+
+        if not dataclasses.is_dataclass(python_val):
```

Comment:
I don't want to reuse the code in `to_literal` and `to_python_val` because I think this will be far more readable in the future when we want to have more flexible change to the JSON IDL object.

Reply:
I haven't come up with the scenario, but this is just my instinct.